### PR TITLE
Fix: Hive fails on linux due to wrong Mutex instantiation

### LINKED ISSUE #60: IOException from CreateMutexCore on Linux
### Expected Behavior

The Xive performs well on unix systems
The repo is testet on linux on the appveyor
### Actual Behavior

IOException when creating a mutex
### The log given by the failure.

System.IO.IOException: The filename, directory name, or volume label syntax is incorrect : 'Local/XC5WS5N0cbr2I_aEvsFwoQ=='
   at System.Threading.Mutex.CreateMutexCore(Boolean initiallyOwned, String name, Boolean& createdNew)
   at Xive.Xocument.MutexXocument.Block()
   at Xive.Xocument.MutexXocument.Nodes(String xpath)
   at Xive.Hive.MutexCatalog.Create(String id)

### PR DESCRIPTION
**Pull request by jaanpeeter**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

Does not work on Linux
Is not testet on Linux
### What is the new behavior?

closes #60 
Implements Linux Appveyor build
### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information
"local/" is removed from mutext name because of: "If you do not specify a prefix when you create a named mutex, it takes the prefix "Local\"."
https://docs.microsoft.com/en-us/dotnet/api/system.threading.mutex?view=netcore-2.2

### CODE REVIEW

File: src/Xive/Hive/MutexHive.cs [hunk 1]
```diff
@@ -108,7 +108,7 @@ private void Block()
                 
                 if (this.mtx.Count == 0)
                 {
-                    this.mtx.Add(new Mutex(false, $"Local/{this.hive.Scope()}"));
```
Comment (Meerownymous) [inline src/Xive/Hive/MutexHive.cs]: I think the correct fix, according to the msdn docs, would be to change the / to a \ - please do this and check if the linux build works.

See: 

> "On a server that is running Terminal Services, a named system mutex can have two levels of visibility. If its name begins with the prefix "Global\", the mutex is visible in all terminal server sessions. If its name begins with the prefix "Local\", the mutex is visible only in the terminal server session where it was created. In that case, a separate mutex with the same name can exist in each of the other terminal server sessions on the server. If you do not specify a prefix when you create a named mutex, it takes the prefix "Local\". Within a terminal server session, two mutexes whose names differ only by their prefixes are separate mutexes, and both are visible to all processes in the terminal server session. That is, the prefix names "Global\" and "Local\" describe the scope of the mutex name relative to terminal server sessions, not relative to processes."

https://docs.microsoft.com/en-gb/dotnet/api/system.threading.mutex?view=netframework-4.8

File: build.cake [hunk 2]
```diff
@@ -108,28 +107,35 @@ Task("Generate-Coverage")
...
+		catch(Exception ex)
+		{
+			Information("Error: " + ex.ToString());
+		}
 	}
-	catch(Exception ex)
+	else
 	{
-		Information("Error: " + ex.ToString());
+Verbose("Is Linux no  coverage");
```
Comment (Meerownymous) [inline build.cake]: Please use indentation
Please use a correct phrasing (like "Generating coverage is not supported on Linux")

File: src/Xive/Hive/MutexHive.cs (hunk 1 above)
Comment (jaanpeeter) [reply src/Xive/Hive/MutexHive.cs]: I tried that. It doesn't work.

**Comment by jaanpeeter** [comment]
> Please check the code for any Mutex instantiations that you changed. There is at least one change to Local/ left which you removed.
> Also, I noticed you replaced strings in tests with Path.DirectorySeperatorChar. If this was necessary to enable Xive to work on other platforms, please do it differently: Change the code so that Xive normalizes all paths - meaning that it replaces the seperator internally and it will always deliver the documents, no matter which seperator a user specifies. This makes Xive more robust against its users calling using wrong paths and getting confused.

Yes I removed all "Local/", because it is the default and does not work under Linux.

File: src/Xive/Hive/FileHive.cs [hunk 3]
```diff
@@ -64,10 +64,7 @@ public FileHive(string scope, string root, Func<IHoneyComb, IHoneyComb> combWrap
             this.scope = scope;
             this.root = new Normalized(root);
             this.wrap = combWrap;
-            this.root = new Solid<string>(() =>
-            {
-                return new Normalized(root).AsString();
```
Comment (Meerownymous) [inline src/Xive/Hive/FileHive.cs]: You removed the stickyness of the value by removing Solid. Please restore the caching.

**Pull request merged**